### DISCUSSION TITLE
[Errno 11001] getaddrinfo failed)

### DISCUSSION THREAD

**Discussion by alexyapsl** [ig-python/ig-markets-api-python-library#344] [Q&A]
`HTTPSConnectionPool(host='demo-api.ig.com', port=443): Max retries exceeded with url: /gateway/deal/history/activity/5000 (Caused by NameResolutionError("<urllib3.connection.HTTPSConnection object at 0x00000244ADAC9300>: Failed to resolve 'demo-api.ig.com' ([Errno 11001] getaddrinfo failed)"))`

Got above error when using API call for transaction activity on an overnight run on DJI on 27th December ( I am based in Asia, so running this for 27th Dec trading hours as I go to sleep). Try to reproduce it today but could not.

Was there a DNS change overnight or any change on  the port or something on demo api that caused this on 27th Dec ?

Timing is around 2024-12-27T14:51:06.351 (GMT)

**Comment by galburn** [comment]
I didn't see any DNS changes from IG at that time, but I had exactly the same error with a different end point on 2023-11-28.  It was not repeatable, but I traced it to my ISP bouncing their DNS server without having a backup server.  I now use a more reliable DNS server.  Hopefully this does not happen often to you, but if it does consider using Google's free public DNS server which is very predictable (8.8.8.8).
